woocommerce/woocommerce-ios
# HACK week: Handle cookie authentication with URLSession

Final part of a POC for HACK week in December 2022.


### Description

This PR continues the work from #8346 to complete the replacement of Alamofire by updating `WordPressOrgNetwork` to use the new `SessionManager` instead.

Specifically, `WordPressOrgNetwork` was updated to keep a reference to a retrieved cookie nonce instead of depending on `WordPressKit`'s `CookieNonceAuthenticator`. Before each request is made, we check if a nonce exists, if not, a request is made to authenticate the user and retrieve the nonce.

Finally, all imports of Alamofire were removed. References to `AFError` were replaced with `NetworkError`.

### Testing instructions

Cookie authentication is currently used only during the Jetpack setup & connection flows.

1. Native Jetpack setup flow
- This flow is currently A/B tested, so to ensure the availability of this flow, disable the [check](https://github.com/woocommerce/woocommerce-ios/blob/4a72b0175d75cf336ed7086dc1951efcac21340f/WooCommerce/Classes/Authentication/AuthenticationManager.swift#L766) for the A/B experiment to always enable it.
- Build the app and log in with your WP.com account.
- On the store picker, select Add a store button.
- On the presented site address screen, enter the address of a test self-hosted store that you have access to and doesn't have Jetpack.
- Enter incorrect credentials to the site. Ensure that you get an appropriate message.
- Enter the correct credentials. Notice that the Jetpack setup screen is presented. Ensure that the process completes.

2. Web Jetpack connection flow
- Log out of the app and select Enter your site address. If you don't see the option, update the [check](https://github.com/woocommerce/woocommerce-ios/blob/4a72b0175d75cf336ed7086dc1951efcac21340f/WooCommerce/Classes/Authentication/AuthenticationManager.swift#L61) for the simplified login flow to always return `false`.
- Enter the address of a test self-hosted site that you have access to. The site should has Jetpack installed and activated but not connected to your WP.com account.
- On the error screen that is presented, select Connect Jetpack.
- Enter the correct credentials to the site.
- Notice that a web view should then be displayed for handling Jetpack connection. This means the Jetpack connection URL has been fetched successfully using cookie authentication.

### Screenshots

N/A

---
